tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ITelemeteredConfigurationSource.cs

```diff
@@ -89,6 +89,31 @@ internal interface ITelemeteredConfigurationSource
     /// <returns>The value of the setting, or <c>null</c> if not found.</returns>
     ConfigurationResult<IDictionary<string, string>>? GetDictionary(string key, IConfigurationTelemetry telemetry, Func<IDictionary<string, string>, bool>? validator, bool allowOptionalMappings);
 
+    /// <summary>
+    /// Gets the <see cref="IDictionary{TKey, TValue}"/> value of
+    /// the setting with the specified key.
+    /// </summary>
+    /// <param name="key">The key that identifies the setting.</param>
+    /// <param name="telemetry">The context for recording telemetry.</param>
+    /// <param name="validator">An optional validation function that must be applied to
+    /// a successfully extracted value to determine if it should be accepted</param>
+    /// <param name="selector">Selector for dictionary</param>
+    /// <returns>The value of the setting, or <c>null</c> if not found.</returns>
+    unsafe ConfigurationResult<IDictionary<string, string>>? GetDictionary(string key, IConfigurationTelemetry telemetry, Func<IDictionary<string, string>, bool>? validator, delegate*<ref string, ref string, bool> selector);
```

Comment:
In performant scenarios it does make a difference, this will be converted to a `calli` instruction, while any delegate is converted to a new type and some virtual calls. Maybe for this scenario it doesn't make sense 🤷🏻  but I'm curious why do you hate it?

Reply:
> Maybe for this scenario it doesn't make sense

Yeah, that was my main thinking, we call this probably what, 5 times? Seems overkill for a such a little-used method..

> but I'm curious why do you hate it?

They're ugly, harder to read, harder to call, are only understood by <1% of .NET developers, and have an un-measured perf benefit 😛